submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/model/DocSubmissionExt.kt

```diff
@@ -36,7 +36,7 @@ val DocSubmission.allDocSections: List<DocSection>
         .filterNotNull()
 
 val DocSection.allSections: List<DocSection>
-    get() = sections.flatMap { either -> either.fold({ it.allSections }, { emptyList() }) }
+    get() = sections.flatMap { either -> either.fold({ listOf(it).plus(it.allSections) }, { emptyList() }) }
```

Comment:
maybe `get() = sections.flatMap { it.fold({ section -> listOf(section) + section.allSections }, { emptyList() }) }`